packages/compass-user-data/src/user-data.ts

```diff
@@ -252,11 +252,230 @@ export class FileUserData<T extends z.Schema> extends IUserData<T> {
   async updateAttributes(
     id: string,
     data: Partial<z.input<T>>
-  ): Promise<z.output<T>> {
-    await this.write(id, {
-      ...((await this.readOne(id)) ?? {}),
-      ...data,
-    });
-    return await this.readOne(id);
+  ): Promise<boolean> {
+    try {
+      await this.write(id, {
+        ...((await this.readOne(id)) ?? {}),
+        ...data,
+      });
+      return true;
+    } catch {
+      return false;
+    }
+  }
+}
+
+// TODO: update endpoints to reflect the merged api endpoints https://jira.mongodb.org/browse/CLOUDP-329716
+export class AtlasUserData<T extends z.Schema> extends IUserData<T> {
+  private readonly authenticatedFetch;
+  private readonly getResourceUrl;
+  private orgId: string = '';
+  private projectId: string = '';
+  constructor(
+    validator: T,
+    dataType: string,
+    orgId: string,
+    projectId: string,
+    getResourceUrl: GetResourceUrl,
+    authenticatedFetch: AuthenticatedFetch,
```

Comment:
When we start getting this many arguments it reads nicer if we include them in an object type. How does it sound if we have all of these except the validator in `AtlasUserDataOptions`?